python/tvm/relay/op/tensor.py

```diff
@@ -616,3 +618,68 @@ def copy(data):
         The copied result.
     """
     return _make.copy(data)
+
+
+def device_copy(data, src_dev, dst_dev):
+    """Copy data from thte source device to the destination device. This
+    operator helps data transferring between difference contexts for
+    heterogeneous execution.
+
+    Parameters
+    ----------
+    data : tvm.relay.Expr
+        The tensor to be copied.
+
+    src_dev : Union(:py:class:`TVMContext`, str)
+        The source device where the data is copied from.
+
+    dst_dev : Union(:py:class:`TVMContext`, str)
+        The destination device where the data is copied to.
+
+    Returns
+    -------
+    result : tvm.relay.Expr
+        The copied result.
+    """
+    if isinstance(src_dev, _TVMContext):
+        src_dev = src_dev.device_type
+    elif isinstance(src_dev, str):
+        src_dev = _nd.context(src_dev).device_type
+    else:
+        raise ValueError("src_dev is expected to be the type of TVMContext or "
+                         "str, but received %s" % (type(src_dev)))
+
+    if isinstance(dst_dev, _TVMContext):
+        dst_dev = dst_dev.device_type
+    elif isinstance(dst_dev, str):
+        dst_dev = _nd.context(dst_dev).device_type
+    else:
+        raise ValueError("dst_dev is expected to be the type of TVMContext or "
+                         "str, but received %s" % (type(dst_dev)))
+    return _make.device_copy(data, src_dev, dst_dev)
+
+
+def on_device(data, device):
+    """Annotate a tensor with device id.
```

Comment:
I think this should be, since we can annotate an entire expression to execute on a different device. 

```suggestion
    """Annotate an expression with device id.
```